File: src/evolution_runtime/runtimeStrategyEngine.ts
/**
 * Converted from Python: core/evolution_runtime/runtime_strategy_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export function buildRuntimeStrategy(evidence: any = null): any {
  evidence = py.or2(evidence, () => ({}));
  var drift_count: any = py.toInt(py.get(evidence, "drift_count", 0));
  var failed_steps: any = py.toInt(py.get(evidence, "failed_steps", 0));
  return {"extraction_path": (py.eq(drift_count, 0) ? "browser_first" : "repair_then_extract"), "synchronization_path": (py.eq(failed_steps, 0) ? "continuous" : "converge_then_sync"), "workflow_order": "priority_asc", "selector_hierarchy": ["healed", "structural", "fallback"], "recovery_order": ["heal_selector", "recover_workflow", "resync_runtime"], "bounded": true};
}

File: lib/src/evolution/runtime_strategy_engine.dart
/// Port of core/evolution_runtime/runtime_strategy_engine.py
Map<String, dynamic> buildRuntimeStrategy([Map<String, dynamic>? evidence]) {
  final e = evidence ?? <String, dynamic>{};
  final driftCount = _asInt(e['drift_count']);
  final failedSteps = _asInt(e['failed_steps']);

  return <String, dynamic>{
    'extraction_path':
        driftCount == 0 ? 'browser_first' : 'repair_then_extract',
    'synchronization_path':
        failedSteps == 0 ? 'continuous' : 'converge_then_sync',
    'workflow_order': 'priority_asc',
    'selector_hierarchy': <String>['healed', 'structural', 'fallback'],
    'recovery_order': <String>[
      'heal_selector',
      'recover_workflow',
      'resync_runtime',
    ],
    'bounded': true,
  };
}

int _asInt(dynamic v) {
  if (v is int) return v;
  if (v is num) return v.toInt();
  return 0;
}
